### PR TITLE
feat: 路透社分站：CN & UK

### LINKED ISSUE #5051: [bug] 部分路透路由 bug
### 网站地址
https://www.reuters.com/
https://cn.reuters.com/

### 网站描述
路透社官方不再支持 RSS feed 
[Confirmed: Reuters is officially killing off RSS support](https://www.reddit.com/r/rss/comments/hf2fjx/confirmed_reuters_is_officially_killing_off_rss/)

### 需要生成什么内容？
不同板块内容，[实时资讯](https://cn.reuters.com/theWire)、[深度分析](https://cn.reuters.com/news/analyses)等

### 额外描述
移动版 http://cn.mobile.reuters.com/ 或许更加好抓取，但不支持https

### PR DESCRIPTION
**Pull request by HenryQW**
涵盖了大多数免费频道。

Close #5051

### CODE REVIEW

File: lib/routes/reuters/channel.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    const response = await got.get(link);
+
+    const $ = cheerio.load(response.data);
+
+    const links = $('.inlineLinks a').map((i, e) => 'https://cn.reuters.com' + e.attribs.href);
+
+    const items = await Promise.all(
+        links.splice(0, 10).map(async (link) => {
+            const single = await utils.ProcessFeed(link);
```
Comment (DIYgod) [inline lib/routes/reuters/channel.js:73]: 应该使用缓存

**Comment by proletarius101** [comment]
Hopefully the router would indicate languages, such that later it's easier to add en-us feeds. Same for #5080

**Comment by HenryQW** [comment]
> Hopefully the router would indicate languages, such that later it's easier to add en-us feeds. Same for #5080

Sites in other language have a very different layout, so the logic will have to be rewritten.

**Comment by proletarius101** [comment]
> Sites in other language have a very different layout, so the logic will have to be rewritten.

Yeah. So if you may write a router like '/reuters/:lang/: category', I'll be able to add a parser for en-us. Reuters has a independent domain (subdomain) for each language. There should be an correspondence beteeen the router in RSSHub and Reuters' website.

**Comment by HenryQW** [comment]
@proletarius101 I've modified it for potential future expansion. However, I'm unable to find out a way to handle slideshows (https://uk.reuters.com/article/uk-minneapolis-police-fashion-campbell/naomi-campbell-sees-black-lives-matter-altering-fashion-and-beauty-industries-idUKKBN2401BV) and videos (https://uk.reuters.com/article/us-health-coronavirus-sweden-bar-app/swedish-drinkers-to-go-the-distance-with-new-pub-app-idUKKBN2411EY) in the article.

**Comment by HenryQW** [comment]
@DIYgod 我的 `yarn format` 不知道为什么跪了

**Comment by proletarius101** [comment]
> I've modified it for potential future expansion. However, I'm unable to find out a way to handle [slideshows](https://uk.reuters.com/article/uk-minneapolis-police-fashion-campbell/naomi-campbell-sees-black-lives-matter-altering-fashion-and-beauty-industries-idUKKBN2401BV) and [videos](https://uk.reuters.com/article/us-health-coronavirus-sweden-bar-app/swedish-drinkers-to-go-the-distance-with-new-pub-app-idUKKBN2411EY) in the article.

Would you push your page parsing code? I don't see it in https://github.com/HenryQW/RSSHub/tree/route/reuters so I can't just give you a pull request.

Probably slideshows will be handled this way:

1.  Find the slideshow's position: `const e = $('.Slideshow_container')`
2.  Use regrex to find slide data stored in a `<script>` : `/(?<=<script type="text/javascript">window.RCOM_Data = )\{.*\}(?=;</script>)/`
3.  Then you may parse that string, insert ``s before `e`, and remove `e`.

Or, you may want to simply remove the original one and inserts everything after the article, if you feel there are too many pictures.

Videos can be handled the in the same way. Their data is captured by the same regex for pictures.

**Comment by HenryQW** [comment]
@proletarius101 The parsing part is in https://github.com/HenryQW/RSSHub/blob/route/reuters/lib/routes/reuters/utils.js

**Comment by proletarius101** [comment]
I realize that routes such as `https://uk.reuters.com/subjects/banks` are quite random. It may not be the best way to deal with channel names that hardcode them. Pages like `https://uk.reuters.com/subjects/banks` are all in the same structure, meaning you may use `'.story-content a'` to select articles in any of them. 

I think we'd better to allow the subscribers to copy the category from Reuters' sitemap, and append to RSSHub's route (e.q. `/reuters/channel/uk/subjects/banks`). Besides, if there is no category specified, should we just grab them all?

P.S. maybe "uk" should not be called a lang, but a country...

**Comment by HenryQW** [comment]
@proletarius101 There are pages with a different layout so the same selector will not work, I only tested these hardcoded channels so far.

The aim is to build the foundations (utils.js) and expand later. If you can PR the parsing rules for slideshows and videos.

**Comment by proletarius101** [comment]
@HenryQW  Yeah I only tested random samples. But I can write a script to test them all if that will be more rubost. I submited a pr here: HenryQW#537

**Comment by HenryQW** [comment]
@proletarius101 I believe the section part for US site is incorrect in your PR

**Comment by HenryQW** [comment]
@proletarius101 I rebased to include your PR https://github.com/HenryQW/RSSHub/pull/537.

@DIYgod 终于可以 merge 了！

**Pull request merged**